### ISSUE TITLE
Low classification model accuracy since v8.3.66

### ISSUE THREAD

**Issue by robenglish** [ultralytics/ultralytics#21512]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

Hi,

I'm getting very low accuracy training results using any ultralytics version >= v8.3.66. Using custom dataset for binary classification with v11n-cls. 

training run results - ultralytics versions 8.3.65,8.3.66,8.3.169

v8.3.65

[results-v8.3.65.csv](https://github.com/user-attachments/files/21431030/results-v8.3.65.csv)
<img width="1200" height="1200" alt="Image" src="https://github.com/user-attachments/assets/31ee4c85-a221-4de2-81e5-b2fb50343e31" />

v8.3.66
[results-8.3.66.csv](https://github.com/user-attachments/files/21431025/results-8.3.66.csv)
<img width="1200" height="1200" alt="Image" src="https://github.com/user-attachments/assets/b324f4bf-4eab-4c70-ad78-d247a24df37d" />

v8.3.169
<img width="1200" height="1200" alt="Image" src="https://github.com/user-attachments/assets/93e9a9a3-3ac3-4521-b10b-50b7fbb80ab6" />
[results-8.3.169.csv](https://github.com/user-attachments/files/21431021/results-8.3.169.csv)

### Environment

(ultra311) bytronic@Bytronic-System0:~$ yolo checks
Ultralytics 8.3.65 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (NVIDIA GeForce RTX 4070 Ti, 12000MiB)
Setup complete ✅ (4 CPUs, 31.3 GB RAM, 605.5/1831.7 GB disk)

OS                  Linux-6.14.0-24-generic-x86_64-with-glibc2.39
Environment         Linux
Python              3.11.12
Install             pip
RAM                 31.27 GB
Disk                605.5/1831.7 GB
CPU                 Intel Core(TM) i5-6600K 3.50GHz
CPU count           4
GPU                 NVIDIA GeForce RTX 4070 Ti, 12000MiB
GPU count           1
CUDA                12.4

numpy               ✅ 1.26.4>=1.23.0
numpy               ✅ 1.26.4<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.9.4>=3.3.0
opencv-python       ✅ 4.11.0.86>=4.6.0
pillow              ✅ 11.1.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.15.2>=1.4.1
torch               ✅ 2.6.0>=1.8.0
torch               ✅ 2.6.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.21.0>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 7.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.14>=2.0.0

### Minimal Reproducible Example

#pip install ultralytics==8.3.65 
#good model accuracy

#pip install ultralytics==8.3.66
#poor model accuracy

#pip install -U ultralytics

<img width="1200" height="1200" alt="Image" src="https://github.com/user-attachments/assets/ecbb363d-f984-4552-918e-69bd1729548f" />
[results-8.3.169.csv](https://github.com/user-attachments/files/21431013/results-8.3.169.csv)

#poor model accuracy (8.3.169)

from ultralytics import YOLO

model = YOLO("yolo11n-cls.pt")

model.train(data="path/to/dataset", epochs=100, imgsz=640)

### Additional

Yolo is great! Thanks!

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @robenglish, thank you for sharing your detailed findings and for your interest in Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will review and assist you further soon.

We recommend checking out the [Docs](https://docs.ultralytics.com/) for the latest guidance on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, as many common issues and training tips are covered there.

Since this is a 🐛 Bug Report, thank you for already providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) and detailed logs! This helps us a lot in identifying the root cause.

If you have additional information (like sample images or more training logs), please share them to further assist our debugging.

Join the Ultralytics community for more support and discussion:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth topics
- [Subreddit](https://www.reddit.com/r/Ultralytics/) to share and learn with others

## Upgrade

To ensure you’re running the latest features and fixes, upgrade your `ultralytics` package (and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can run in any of the following up-to-date, verified environments with all dependencies (including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/)) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI covers all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your valuable feedback and for helping improve Ultralytics!

**Comment by Y-T-G** [comment]
How much data do you have?

**Comment by Y-T-G** [comment]
From what I can see, [this PR](https://github.com/ultralytics/ultralytics/pull/18842) fixed the automatic LR calculation. Classification models were always using `nc=10` because `model.nc` wasn't an attribute for classification models. So after the change, the LR is being calculated with the correct `nc` which now results in a higher LR for binary classification.

If you want to still use the old LR, you can pass `lr0=0.000714, optimizer="AdamW" momentum=0.9 warmup_bias_lr=0.0`.

This only affects training with small datasets.

@Laughing-q
